src/reconcile/rec_col.c

```diff
@@ -323,9 +342,16 @@ __wt_rec_col_int(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REF *pageref)
         if (page_del != NULL)
             WT_TIME_AGGREGATE_MERGE(session, &r->cur_ptr->ta, &ft_ta);
         WT_TIME_AGGREGATE_MERGE(session, &r->cur_ptr->ta, &ta);
+        WT_PAGE_STAT_UPDATE(&r->cur_ptr->ps, &ps);
     }
     WT_INTL_FOREACH_END;
 
+    /* Reset the page stat value for the page if any of its children had invalid stat values. */
```

Comment:
Resetting here can be wrong. We already have split the page above. Instead of storing the flag and resetting it later,  reset there itself inside the `WT_PAGE_STAT_UPDATE` macro and do not update further if the reset flag is set.

Reply:
Ok, thanks for the suggestion.